### PR TITLE
feat: Include scanner output in Issue body

### PR DESCRIPTION
**Pull request by jwatson-ccc**

### CODE REVIEW

**Comment by jwatson-ccc** [comment]
@mmcgehee-ccc 
This is still a bit of a work in progress, but it is working:
- It includes the reports in the issue body
- It will update the issue with the latest report on a rerun

Todo:
- Decide if we want to change the Slack message. 
- Maybe change to link to the Issues?
- Add links to the issue?

**Comment by mmcgehee-ccc** [comment]
@jwatson-ccc 

If the govulncheck or trivy job fails to run (e.g., trivy job fails to run due to docker build failure), then an issue will **_not_** be created/updated for the failure. However, I think this is acceptable since a Slack alert will still occur to notify us of the problem. Do you agree?

File: .github/workflows/golang-security-scan.yml [hunk 1]
```diff
@@ -66,63 +94,118 @@ jobs:
...
+            if [[ "${{ matrix.label }}" == "govulncheck" ]]; then
+              ISSUE_TITLE="Go Vulnerability Report"
+              FILE_PATH=govulncheck-report.txt
+            elif [[ "${{ matrix.label }}" == "trivy" ]]; then
+              ISSUE_TITLE="Trivy Vulnerability Report"
+              FILE_PATH=trivy-report.txt
+            fi
+
+            echo "ISSUE_TITLE=$ISSUE_TITLE" >> $GITHUB_ENV
+            echo "issue_title=$ISSUE_TITLE" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (jwatson-ccc) [reply .github/workflows/golang-security-scan.yml:154]: I was attempting to us `${{ $ISSUE_TITLE }}`, which was not working.  I didn't know about `${{ env.ISSUE_TITLE }}`

I will give it a shot.

**Comment by jwatson-ccc** [comment]
> @jwatson-ccc
> 
> If the govulncheck or trivy job fails to run (e.g., trivy job fails to run due to docker build failure), then an issue will **_not_** be created/updated for the failure. However, I think this is acceptable since a Slack alert will still occur to notify us of the problem. Do you agree?

It looks to me that the only reason the issue creation would fail is because $FILE_PATH would be pointing to a non-existent file.

Are you seeing something more?

**Comment by mmcgehee-ccc** [comment]
@jwatson-ccc 
I was seeing the `create_issues` job fail at the `actions/download-artifact@v2` step since no artifact named 'trivy' exists if the `Build docker image` step fails.

File: .github/workflows/golang-security-scan.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jwatson-ccc) [reply .github/workflows/golang-security-scan.yml:154]: It looks like it works, but the linter complains.

But I tell myself, we can let the linters drive our code solutions...

image

**Comment by mmcgehee-ccc** [comment]
@jwatson-ccc @zredinger-ccc 
The latest batch of updates for this are complete and ready for review. The Slack message now also includes links to the relevant issues (see the ccc-dev-test Slack channel).

**Comment by jwatson-ccc** [comment]
@mmcgehee-ccc 

I made the following changes:
- Removed matching on the title
- Used the most recent issue if multiple issues are found
- Preserve the existing Title when updating an issue.

I looked over your changes and ran some tests. Everything looks good, so I will put the review back in your court since this is my PR, and I can not approve it.

**Review by mmcgehee-ccc — APPROVED** [review]
LGTM

**Pull request merged**